### ISSUE TITLE
[Bug/Question] Builder don't respect inventory order?

### ISSUE THREAD

**Issue by ReiDaTecnologia** [McJtyMods/RFTools#1883]
Issue description: 
Hello, i'm trying to automate CompactMachines and for it i'm using the Builder with a Scan Shape Card, but i'm having a annoying issue...
In my Test World, when my AE2 put the items in the chest (For the compactwall recipe) sometimes the Builder put the Redstone first breaking all the automation and also doing a really weird sound, like, it try to place the redstone and the iron at the same? but as it can't, it just keep trying until is out of redstone, this is destroying my automation, i really don't know another way to automate it 😭 
[Screenshot](https://user-images.githubusercontent.com/30323772/49360852-87aa3500-f6c1-11e8-8b73-369fd37ed049.png)
Steps to reproduce: 
Do a AE2 system or RS (maybe it will have the same issue) and do a pattern for the Interface in the chest above the Builder, do a scan of the CompactWall recipe using the Scanner and put it in the builder and request it, maybe you will need a few trys (Hopefully will happen with you because maybe i'm just unlucky xD) and you will see what i saw :/

Versions:
* Minecraft: All The Mods 3 Remix (1.12.2 Modpack) - 1.0.1
* Forge: 14.23.5.2768
* CompatLayer (only if on Minecraft 1.10 or 1.11): 1.12.2
* McJtyLib: 1.12-3.0.6
* RFTools: 1.12-7.56

Possibly incompatible mods and versions: 
OptiFine.

Relevant logs, if any: 
N/A (Tell me if you need it)

**Comment by McJty** [comment]
I don't fully understand what you are doing but if the chest only has redstone then it is unavoidable that the builder will put that redstone first since it has nothing else it can do. Even if the supporting blocks aren't there. You have to make sure all needed materials are in the chest before building

**Comment by ReiDaTecnologia** [comment]
Yes, the chest have the needed items but for some reason sometimes it put the redstone first and as the redstone is in the same location of the iron block it can't place it, even putting the iron block in the first slot and the redstone in 2 slot for some reason it use the redstone instead of the iron block, mayble is because the builder is always trying to place it? and in the time that i request, it is checking the 2 slot?

**Comment by McJty** [comment]
Redstone at the same location of the iron block? That's not possible. Redstone is a block in itself and you cannot place it in the same location as another block. I'm still very confused about what you're trying to do though

**Comment by ReiDaTecnologia** [comment]
I will do a video really quick, let me launch the game.

**Comment by ReiDaTecnologia** [comment]
https://youtu.be/r2v2ykeCL6g
Here is the video.
As you can see, in the first try it work as it should but in the next one don't.
I show as well my automation setup, redstone system to activate the builder only when the first crafting is complete (The miniaturization send redstone signals depending the % of completion, 15 being 99%).
I hope it is enough :/

**Comment by Famous5000** [comment]
@ReiDaTecnologia What could be happening is that the builder block is unable to place it properly because of where it is on the scan. It scans from top-bottom, and what you could try doing is doing a redstone pulse after the items are inserted so that the builder will re-scan. That might solve your problem, though if you're able to do a compound "scan" then that may help a bit better.
